### PR TITLE
Task 9 : Tax Calculator

### PR DESCRIPTION
**Pull request by blazejbujala**

### CODE REVIEW

File: app/src/test/groovy/pl/futurecollars/invoicing/services/TaxCalculatorTest.groovy [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    def entry2 = new InvoiceEntry("Skarpetki", 10 as BigDecimal, Vat.VAT_8)
+    def entry3 = new InvoiceEntry("Klapki", 50 as BigDecimal, Vat.VAT_23)
+    def entries1 = Arrays.asList(entry1, entry2)
+    def entries2 = Arrays.asList(entry1, entry3)
+    def entries3 = Arrays.asList(entry1, entry2, entry3)
+    def invoice = new Invoice(UUID.randomUUID(), dateOfIssue1, issuer, receiver, entries1)
+    def invoice2 = new Invoice(UUID.randomUUID(), dateOfIssue2, issuer, receiver2, entries2)
+    def invoice3 = new Invoice(UUID.randomUUID(), dateOfIssue2, issuer2, receiver3, entries3)
+    def repository = new InMemoryRepository()
+    InvoiceService invoiceService = new InvoiceService(repository)
```
Comment (radoslawdabrowski) [inline app/src/test/groovy/pl/futurecollars/invoicing/services/TaxCalculatorTest.groovy:32]: Inject `InvoiceService` as spring bean

File: app/src/test/groovy/pl/futurecollars/invoicing/services/TaxCalculatorTest.groovy [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    def entry1 = new InvoiceEntry("Buty", 100 as BigDecimal, Vat.VAT_23)
+    def entry2 = new InvoiceEntry("Skarpetki", 10 as BigDecimal, Vat.VAT_8)
+    def entry3 = new InvoiceEntry("Klapki", 50 as BigDecimal, Vat.VAT_23)
+    def entries1 = Arrays.asList(entry1, entry2)
+    def entries2 = Arrays.asList(entry1, entry3)
+    def entries3 = Arrays.asList(entry1, entry2, entry3)
+    def invoice = new Invoice(UUID.randomUUID(), dateOfIssue1, issuer, receiver, entries1)
+    def invoice2 = new Invoice(UUID.randomUUID(), dateOfIssue2, issuer, receiver2, entries2)
+    def invoice3 = new Invoice(UUID.randomUUID(), dateOfIssue2, issuer2, receiver3, entries3)
+    def repository = new InMemoryRepository()
```
Comment (radoslawdabrowski) [inline app/src/test/groovy/pl/futurecollars/invoicing/services/TaxCalculatorTest.groovy:31]: Inject `Repository` as spring bean

**Review by radoslawdabrowski — CHANGES_REQUESTED** [review]

File: app/src/test/groovy/pl/futurecollars/invoicing/services/TaxCalculatorTest.groovy (hunk 1 above)
Comment (blazejbujala) [reply app/src/test/groovy/pl/futurecollars/invoicing/services/TaxCalculatorTest.groovy]: done

File: app/src/test/groovy/pl/futurecollars/invoicing/services/TaxCalculatorTest.groovy (hunk 2 above)
Comment (blazejbujala) [reply app/src/test/groovy/pl/futurecollars/invoicing/services/TaxCalculatorTest.groovy]: done

**Review by radoslawdabrowski — APPROVED** [review]

**Review by stanleyszysz — APPROVED** [review]

**Pull request merged**